### PR TITLE
tbb: enabled tbbbind on linux platform

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Second attempt for https://github.com/Homebrew/homebrew-core/pull/112805, but now enabled only for Linux.

### CODE REVIEW

**Review by carlocab — DISMISSED** [review]
To avoid a repeat of the last time you tried this, I'm going to insist that we have a test for tbbbind in the formula. Even a very minimal one suffices.

See

https://github.com/Homebrew/homebrew-core/blob/c863ee4eacdd3f2fe4f34906903ebfb56306e878/Formula/tbb.rb#L19

File: Formula/tbb.rb
Comment (ilya-lavrenov) [reply Formula/tbb.rb]: removed, thanks

File: Formula/tbb.rb
Comment (SMillerDev) [inline Formula/tbb.rb:89]: This doesn't test if it works, it just tests if it exists

Comment (ilya-lavrenov) [reply Formula/tbb.rb]: added more comprehensive test.

**Comment by ilya-lavrenov** [comment]
> To avoid a repeat of the last time you tried this, I'm going to insist that we have a test for tbbbind in the formula. Even a very minimal one suffices.
> 
> See
> 
> https://github.com/Homebrew/homebrew-core/blob/c863ee4eacdd3f2fe4f34906903ebfb56306e878/Formula/tbb.rb#L19

thanks for noting this. Added the test to cover current behavior

File: Formula/tbb.rb
Comment (carlocab) [inline Formula/tbb.rb:155]: Let's not touch whitespace in the embedded patch, please. (Yes, I know it was probably your editor that ate this.)

File: Formula/tbb.rb
Comment (carlocab) [inline Formula/tbb.rb:100]: Perhaps we can `return` these values instead of printing them to stdout? That way we can just do
```ruby
system "./core-types"
```
instead of reading the output with `assert_equal`.

**Review by carlocab — COMMENTED** [review]
Much better, thanks. A few comments.

File: Formula/tbb.rb
Comment (ilya-lavrenov) [reply Formula/tbb.rb]: great suggestion, updated.

**Comment by ilya-lavrenov** [comment]
Anything else I should update to get it merged?

**Comment by ilya-lavrenov** [comment]
Anything else I should update to get it merged?

**Review by SMillerDev — DISMISSED** [review]

**Comment by ilya-lavrenov** [comment]
Hi @carlocab 
Could you please have a look?

**Comment by ilya-lavrenov** [comment]
@SMillerDev should we wait for review from @carlocab or we can merge? It may take a while..

**Comment by SMillerDev** [comment]
Preferably, yes. People used tbb without this for a while, they can wait for a review.

**Comment by ilya-lavrenov** [comment]
> Preferably, yes. People used tbb without this for a while, they can wait for a review.

Ok, let's try to wait :)

Actual reasons why users have not required this functionality earlier:
- Users did not have a demand to use TBBBind, because Intel did not have hybrid processors earlier.
- This functionality is added 1.5 years ago.

**Comment by iMichka** [comment]
Ping @carlocab, can we ship the change? I want to avoid having to dismiss/override your initial review. Change looks good to me

**Review by iMichka — DISMISSED** [review]

**Comment by carlocab** [comment]
Honestly not keen on this, since it seems like only at most two people are interested in the feature, but if you're happy that's fine with me.

**Comment by ilya-lavrenov** [comment]
> ⚠️ Bottle publish [failed](https://github.com/Homebrew/homebrew-core/actions/runs/4063151611).

rebased to rebuild with TBB 2021.8 version.

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**
